### PR TITLE
ft: #184361217 add update password feature for user in case they want to.

### PR DESCRIPTION
**Pull request by musllim**
# this is different from resetting the password they have forgot.

## Description

GIVEN user is validly registered and account is validated. WHEN user/frontend provides old password, new password and confirm password fielated, or rejected due to the user ds THEN users new password should be hashed and updated, or rejected due to the user not remembering old password or new password and confirm password fields not matching.

This feature is fully documented and includes all of it's unit testing.
Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shyakadavis — APPROVED** [review]
LGMT

**Review by SergeInGithub — APPROVED** [review]
LGTM

File: src/controllers/updatePassword.controller.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export const updatePassword = async (req: Request, res: Response) => {
+  try {
+    const { currentPassword, newPassword, newPasswordConfirmation } = req.body;
+    if (
+      currentPassword.length < 8 ||
+      newPassword.length < 8 ||
+      newPasswordConfirmation.length < 8
+    ) {
+      throw new Error("password is vey short, try more than 8 characters");
+    }
```
Comment (shyakadavis) [inline src/controllers/updatePassword.controller.ts:14]: There is already a regex in the user.middleware.ts which you can just place in the route handler. No need for this.

Comment (musllim) [reply src/controllers/updatePassword.controller.ts:14]: thanks, that is awesome I've looked at it

File: src/routes/auth.routes.ts [hunk 2]
```diff
@@ -7,25 +7,28 @@ import {
...
-authRouter.post('/auth/resetPassword/:token', resetPasswordController); // Reset the password
+authRouter.post("/auth/resetPassword/:token", resetPasswordController); // Reset the password
 
-authRouter.post('/auth/2fa', create2FAToken); // Create a 2FA token
+authRouter.post("/auth/2fa", create2FAToken); // Create a 2FA token
 
-authRouter.post('/auth/2fa/verify2FAToken', verify2FAToken); // Verify the 2FA token
+authRouter.post("/auth/2fa/verify2FAToken", verify2FAToken); // Verify the 2FA token
+
+authRouter.post("/updatepassword", isAuth, updatePassword);
```
Comment (shyakadavis) [inline src/routes/auth.routes.ts:32]: You forgot to use the middleware here on the route. 😄 
There is currently only the isAuth, try adding `ValidateJoi(UserSchema.passwordUpdate`... 🔒 🔑

Comment (musllim) [reply src/routes/auth.routes.ts:32]: that is not even working it is in other task don't worry about it

**Review by shyakadavis — DISMISSED** [review]
LGMT

**Review by emilenfc — DISMISSED** [review]
LGMT

**Review by shyakadavis — APPROVED** [review]
LGMT 🏄🏽‍♂️

**Review by emilenfc — APPROVED** [review]

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**